core/tests/snapshot-recovery-test/tests/snapshot-recovery.test.ts

```diff
@@ -0,0 +1,255 @@
+        let reorgDetectorSucceeded = false;
+        try {
+            const rl = readline.createInterface({
+                input: enProcess.stdout,
+                crlfDelay: Infinity
+            });
+
+            // TODO: use a more reliable method to detect recovery success (e.g., based on health checks)
+            for await (const line of rl) {
+                if (IMPORTANT_LINE_REGEX.test(line)) {
+                    console.log('en> ' + line);
+                }
+                await fs.appendFile(logs, line + '\n');
+
+                if (/L1 batch #\d+ is consistent with L1/.test(line)) {
+                    console.log('Consistency checker successfully checked post-snapshot L1 batch');
+                    consistencyCheckerSucceeded = true;
+                }
+                if (/No reorg at L1 batch #\d+/.test(line)) {
+                    console.log('Reorg detector successfully checked post-snapshot L1 batch');
+                    reorgDetectorSucceeded = true;
+                }
+
+                if (consistencyCheckerSucceeded && reorgDetectorSucceeded) {
+                    break;
+                }
+            }
+
+            // If `enProcess` fails early, we'll trip these checks.
+            expect(enProcess.exitCode).to.be.null;
+            expect(consistencyCheckerSucceeded, 'consistency check failed').to.be.true;
+            expect(reorgDetectorSucceeded, 'reorg detection check failed').to.be.true;
+        } finally {
+            enProcess.kill();
+            await killExternalNode();
+        }
+    });
+});
+
+async function waitForProcess(childProcess: ChildProcess) {
+    await new Promise((resolve, reject) => {
+        childProcess.on('error', (error) => {
+            reject(error);
+        });
+        childProcess.on('exit', (code) => {
+            if (code === 0) {
+                resolve(undefined);
+            } else {
+                reject(new Error(`Process exited with non-zero code: ${code}`));
+            }
+        });
+    });
+}
+
+async function decompressGzip(filePath: string): Promise<Buffer> {
+    const readStream = (await fs.open(filePath)).createReadStream();
+    return new Promise((resolve, reject) => {
+        const gunzip = zlib.createGunzip();
+        let chunks: Uint8Array[] = [];
+
+        gunzip.on('data', (chunk) => chunks.push(chunk));
+        gunzip.on('end', () => resolve(Buffer.concat(chunks)));
+        gunzip.on('error', reject);
+        readStream.pipe(gunzip);
+    });
+}
+
+async function killExternalNode() {
+    interface ChildProcessError extends Error {
+        readonly code: number | null;
+    }
+
+    try {
+        await promisify(exec)('killall -q -KILL zksync_external_node');
+    } catch (err) {
+        const typedErr = err as ChildProcessError;
+        if (typedErr.code === 1) {
+            // No matching processes were found; this is fine.
+        } else {
+            throw err;
+        }
+    }
+}
```

Comment:
honestly, I thought it will be even more bothersome to do this test, so, well done!